R-package/R/xgb.Booster.R

```diff
@@ -1,180 +1,88 @@
-# Construct an internal xgboost Booster and return a handle to it.
+# Construct an internal xgboost Booster and get its current number of rounds.
 # internal utility function
-xgb.Booster.handle <- function(params, cachelist, modelfile, handle) {
+# Note: the number of rounds in the C booster gets reset to zero when updating
+# the parameters through 'xgb.parameters', hence the need to return it from
+# this function when using it for training continuation
+xgb.Booster <- function(params, cachelist, modelfile, training_continuation) {
```

Comment:
I'm not sure about the necessity of `training_continuation` parameter.

> It's now also possible to incrementally update a booster in-place through xgb.trai

Making a copy is not too bad, it conforms to the R semantic of immutable data. A single copy is quite cheap compared to the training itself. You can verify this by generating large models and serializing with UBJSON in memory (not writing to file).

Edit: it seems you are copying the booster anyway by calling a `duplicate` function?

Reply:
Removed.

It was BTW only duplicating the R attributes at the end, not the whole object.